jumanji/environments/combinatorial/cvrp/utils.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from chex import Array, PRNGKey
+from jax import numpy as jnp
+from jax import random
+from jumanji.environments.combinatorial.tsp.utils import get_augmentations as get_augmentations_tsp
+
+DEPOT_IDX = 0
+MIN_NORM_FACTOR = 10
+
+
+def compute_tour_length(problem: Array, order: Array) -> jnp.float32:
+    """Calculate the length of a tour."""
+    problem = problem[order, 0:2]
+    return jnp.linalg.norm((problem - jnp.roll(problem, -1, axis=0)), axis=1).sum()
+
+
+def generate_problem(problem_key: PRNGKey, num_nodes: jnp.int32) -> Array:
+    coords = random.uniform(problem_key, (num_nodes + 1, 2), minval=0, maxval=1)
+    costs = random.randint(problem_key, (num_nodes + 1, 1), minval=1, maxval=10)
+    problem = jnp.hstack((coords, costs))
+    problem = problem.at[DEPOT_IDX, 2].set(0.0)
+    return problem
```

Comment:
Let's avoid using the same key for two random samplings. 

```suggestion
def generate_problem(key: PRNGKey, num_nodes: jnp.int32) -> Array:
    coord_key, cost_key = jax.random.split(key)
    coords = random.uniform(coord_key, (num_nodes + 1, 2), minval=0, maxval=1)
    costs = random.randint(cost_key, (num_nodes + 1, 1), minval=1, maxval=10)
    problem = jnp.hstack((coords, costs))
    problem = problem.at[DEPOT_IDX, 2].set(0.0)
    return problem
```